org.eclipse.scanning.test/src/org/eclipse/scanning/test/scan/SeekTest.java

```diff
@@ -0,0 +1,146 @@
+package org.eclipse.scanning.test.scan;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertNotEquals;
+import static org.junit.Assert.assertNotNull;
+import static org.junit.Assert.assertTrue;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.concurrent.TimeUnit;
+
+import org.eclipse.scanning.api.device.AbstractRunnableDevice;
+import org.eclipse.scanning.api.device.IDeviceController;
+import org.eclipse.scanning.api.points.IPosition;
+import org.eclipse.scanning.api.scan.PositionEvent;
+import org.eclipse.scanning.api.scan.ScanningException;
+import org.eclipse.scanning.api.scan.event.IPositionListener;
+import org.eclipse.scanning.api.scan.models.ScanModel;
+import org.junit.Before;
+import org.junit.Ignore;
+import org.junit.Test;
+
+public class SeekTest extends AbstractAcquisitionTest {
+
+	@Before
+	public void setupServices() throws Exception {
+		super.setupServices();
+	}
+
+	@Test
+	public void seekFirst() throws Exception {
+		
+		IDeviceController controller = createTestScanner(null);
+		AbstractRunnableDevice<ScanModel> scanner = (AbstractRunnableDevice<ScanModel>)controller.getDevice();
+    
+		try {
+			scanner.start(null);		
+			Thread.sleep(100);	
+			scanner.pause();
+			
+			IPosition first   = scanner.getModel().getPositionIterable().iterator().next();
+			IPosition current = scanner.getPositioner().getPosition();
+			assertNotNull(current);
+			assertNotEquals(first, current);
+			
+			scanner.seek(0);
+			current = scanner.getPositioner().getPosition();
+			assertNotNull(current);
+			assertEquals(first.getStepIndex(), current.getStepIndex());
+			
+		} finally {
+			scanner.abort();
+		}
+	}
+	
+	@Ignore("This MUST pass but requires a larger refactor so seeing that the initial changes do with the tests first.")
+	@Test
+	public void seekFirstRestartsInCorrectLocation() throws Exception {
+		
+		IDeviceController controller = createTestScanner(null);
+		AbstractRunnableDevice<ScanModel> scanner = (AbstractRunnableDevice<ScanModel>)controller.getDevice();
+		
+		final List<Integer> steps = new ArrayList<Integer>();
+		scanner.addPositionListener(new IPositionListener() {
+			public void positionPerformed(PositionEvent evt) throws ScanningException {
+				steps.add(evt.getPosition().getStepIndex());
+			}
+		});
+		try {
+			scanner.start(null);		
+			Thread.sleep(100);	

```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR') Remove this use of "Thread.sleep()". [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AS2925)